x/m3em/node/node.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/m3db/m3em/node"
+
+	m3dbrpc "github.com/m3db/m3db/generated/thrift/rpc"
+	m3dbchannel "github.com/m3db/m3db/network/server/tchannelthrift/node/channel"
+	tchannel "github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+)
+
+type m3dbNode struct {
+	sync.Mutex
+	node.ServiceNode
+
+	opts       Options
+	m3dbClient m3dbrpc.TChanNode
+}
+
+// New constructs a new M3DBNode
+func New(svcNode node.ServiceNode, opts Options) (Node, error) {
+	if err := opts.Validate(); err != nil {
+		return nil, err
+	}
+	return &m3dbNode{
+		ServiceNode: svcNode,
+		opts:        opts,
+	}, nil
+}
+
+func (n *m3dbNode) thriftClient() (m3dbrpc.TChanNode, error) {
```

Comment:
Can we just call this `client()`?  If we can eventually get us migrated to GRPC then we don't have to rename methods we might miss when we do the big bang switch.  i.e. I don't want to do an automatic rename and have a `thriftClient()` method returning a GRPC or even agnostic RPC type client, hehe.